### ISSUE TITLE
Incorrect attribute assignment for bracketed text within code within items

### ISSUE THREAD

**Issue by rivy** [arve0/markdown-it-attrs#94]
I came across this bug while writing a README for a javascript package.

This test fails (see https://github.com/rivy/js.markdown-it-attrs/commit/795a537a53b54ec2a877e2d710c3cc66c0eede9e in [rivy/js.markdown-it-attrs/fix.item+code](https://github.com/rivy/js.markdown-it-attrs/tree/fix.item+code))...

```
    it(replaceDelimiters('should not apply inside item lists with `code{.red}`', options), () => {
      src = '- item with `code = {.red}`';
      expected = '<ul>\n<li>item with <code>code = {.red}</code></li>\n</ul>\n';
      assert.equal(md.render(replaceDelimiters(src, options)), replaceDelimiters(expected, options));
    });
```

The bracketed text is being removed from the code section and applied as an attribute to the item. It should, I believe, stay within the code section, untouched.

Example input:
```md
- item with `code = {.red}`
```

Current output:
```html
<ul>
<li class="red">item with <code>code =</code></li>
</ul>
```

Expected output:
```html
<ul>
<li>item with <code>code = {.red}</code></li>
</ul>
```

**Comment by arve0** [comment]
Thanks for the report🙂 I’m on vacation and will be away from computer until next week. 

Feel free to look at the token tree(markdow-it.github.io, choose debug) and the rules (patterns.js) to figure out what fails.

**Comment by rivy** [comment]
With your hints, I was able to narrow down the exact issue, added some tests, and a PR with a fix (#95).

Note that the first two commits of the PR fix this specific issue. I added two more tests that I thought were related in the third commit (which pass without any code changes), and the final commit is a judgement call. I believe the `content: utils.hasDelimiters('only', options)` requirement makes the addition of `type: text` unnecessary, but it fit thematically with the same code fix previously.

I'm happy to edit out the last or the last two commits if you think that would be best.

Let me know if you'd like any changes *after* you get back from your vacation.
🏖🌴🍻 😃
